gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/FilterOptimizingVisitor.java

```diff
@@ -155,7 +155,20 @@ public QueryNode visit(SortAndPaginateNode sortAndPaginateNode) {
                               .addAllChildFilter(filterList)
                               .build();
                     }
-                    return new DataFetcherNode(stringListEntry.getKey(), filter);
+                    // There should always be at least one entry
+                    if (!stringListEntry.getValue().isEmpty()) {
+                      String source = stringListEntry.getKey();
+                      DataFetcherNode dataFetcherNode = (DataFetcherNode) stringListEntry.getValue().get(0);
+                      return
+                          new DataFetcherNode(
+                              source,
+                              filter,
+                              dataFetcherNode.getLimit(),
+                              dataFetcherNode.getOffset(),
+                              dataFetcherNode.getOrderByExpressionList());
+                    } else {
+                      return new NoOpNode();
```

Comment:
Test for this negative scenario?

Reply:
The negative scenario should never occur, but I can still add a unit test for the case

Reply:
Can't even add a unit test, since the case will never occur